pandas/io/pytables.py

```diff
@@ -2558,16 +2586,20 @@ def validate_names(self):
             # TODO: should the message here be more specifically non-str?
             raise ValueError("cannot have non-object label DataIndexableCol")
 
-    def get_atom_string(self, shape, itemsize):
+    @classmethod
+    def get_atom_string(cls, shape, itemsize):
         return _tables().StringCol(itemsize=itemsize)
 
-    def get_atom_data(self, shape, kind: str) -> "Col":
-        return self.get_atom_coltype(kind=kind)()
+    @classmethod
```

Comment:
why class methods? can we not use free functions?

Reply:
because we have two classes that implement different versions of some of these functions